File: leetcode/easy/682-baseball-game.js
var calPoints = function(operations) {
    total = 0;
    scoreboard = [];
    for(let i = 0; i < operations.length; i++) {
        if(operations[i] === "+") {
            addition = scoreboard[scoreboard.length - 1] + scoreboard[scoreboard.length - 2];
            scoreboard.push(addition);
            total += addition;
        } else if(operations[i] === "C") {
            total -= scoreboard.pop();

        } else if(operations[i] === "D") {
            scoreboard.push(scoreboard[scoreboard.length - 1] * 2);
            total += scoreboard[scoreboard.length - 1];
        } else {
            scoreboard.push(Number(operations[i]));
            total += Number(operations[i]);
        }
    }
    return total
};

File: leetcode/easy/682-baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        total = 0
        scoreboard = []
        for i in range(len(operations)):
            if operations[i] == "+":
                addition = scoreboard[len(scoreboard) - 1] + scoreboard[len(scoreboard) - 2]
                scoreboard.append(addition)
                total += addition
            elif operations[i] == "C":
                total -= scoreboard.pop()
            elif operations[i] == "D":
                scoreboard.append(scoreboard[len(scoreboard) - 1] * 2)
                total += scoreboard[len(scoreboard) - 1]
            else:
                scoreboard.append(int(operations[i]))
                total += int(operations[i])
        return total
